Firestore/Example/FuzzTests/FSTFuzzTestsPrincipal.mm

```diff
@@ -92,6 +96,12 @@ int RunFuzzTestingMain() {
       fuzzer_function = fuzzing::FuzzTestDeserialization;
       break;
 
+    case FuzzingTarget::kFieldPath:
```

Comment:
Because the lambda would have to be able to access the object without capturing it (a capturing lambda cannot be converted to a plain function pointer).

Reply:
Nothing's going to be real clean here since we're interfacing with a c-style function pointer.

For ease of reference, here's the solutions above:
1) Switch on type. (Mina's original)
2) Global pointer. (Costa's first alternative)
3) Move invocation of libfuzzer into FuzzTarget::Fuzz (Costa's second alternative)

Sol 1 seems reasonable and has the advantage of being dead simple. Sol 2 also seems ok. (I don't particularly prefer or not prefer it over Sol 1.) I'm not wild about Sol 3, since I think it's too complicated for what we're trying to do. (Not that it's overly complicated, but since what we're trying to do is simple, the solution ideally should be too.)

Here's another alternative:

4) I had a quick look at libFuzzer. It seems like the normal way to use it is to define LLVMFuzzerTestOneInput and use that (rather than calling fuzzer::FuzzerDriver with a function pointer). I feel like I might be treading over well-trodden ground, but could we create multiple binaries (one per fuzz target) instead of a single binary that switches? It *seems* (based on 5 seconds of searching, so I really, really could be wrong) the more canonical approach to using libfuzzer?

Reply:
I guess let's leave as is, for now at least.

Re. 4 -- I remember we tried that and there were some weird (or so they seemed at the time) linking errors in XCode as it was trying to compile and link libFuzzer (as a pod).